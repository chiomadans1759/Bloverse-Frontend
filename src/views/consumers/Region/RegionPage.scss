@import 'sass-rem';

.logo {
  height: rem(40px);
  padding-left: rem(20px);
}
.top {
    background: white;
    padding: rem(10px);
}
.top-left-container {
  margin-left: 5%;
}
.top-left-container > a {
  color: #000;
}
.top-left-container > a > svg {
  vertical-align: middle;
}
.top-left-container > a > span {
  padding-left: rem(10px);
}
.content{
  h3{
    font-size: rem(18px);
    color: #353535;
    margin-bottom: 9px;
  }
  h5{
    font-size: rem(14px);
    color: #868686;
  }
  .card{
    margin-bottom: rem(14px);
    img{
      height: rem(220px);
      width: 100%;
      margin-top: rem(-38px);
      margin-bottom: rem(-29px);
    }
   p {
      color: #000000;
      font-size: rem(14px);
      padding: rem(1px) rem(8px);
      margin-bottom: rem(5px);
  }

  }
}
.highlight:before {
    /* content: ''; */
    background: #096DD9;
    z-index: 200;
    position: absolute;
    right: 28px;
    padding: 6px 11px;
    color: #fff;
    font-size: 19px;
    border-radius: 50%;
    margin-top: 5%;
    content: "\f00c";
    font-family: "Font Awesome 5 Pro";
}