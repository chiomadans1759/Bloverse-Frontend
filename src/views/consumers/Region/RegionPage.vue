<template>
  <div class="region-container">
    <div class="top">
      <div>
        <span class="top-left-container">
          <a href="/">
            <i class="fal fa-long-arrow-left fa-2x"></i>
            <span>Back</span>
          </a>
          <router-link to="/" class="router-link">
            <img class="logo" src="@/assets/Logo.svg" />
          </router-link>
        </span>
      </div>
    </div>
    <div class="content">
      <div class="container">
        <div class="row">
          
          <div class="col-md-2"></div>
          <div class="col-12 col-md-8 col-sm-8 col-lg-8">
            <h3>Select Your Favourite Country</h3>
            <h5>Selected <b>{{countries.length}}</b> countries</h5>
            <br />
            <div class="row">
              <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/UK.svg" @click="addToCountry('UK')" alt="">
                  <p>UK</p>
                </div>
              </div>
              <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/france.svg" @click="addToCountry('france')" alt="">
                  <p>France</p>
                </div>
              </div>
               <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/canada.svg" @click="addToCountry('canada')" alt="">
                  <p>Canada</p>
                </div>
              </div>
               <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/germany.svg" @click="addToCountry('germany')" alt="">
                  <p>Germany</p>
                </div>
              </div>
               <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/ghana.svg" @click="addToCountry('ghana')" alt="">
                  <p>Ghana</p>
                </div>
              </div>
               <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/australia.svg" @click="addToCountry('australia')" alt="">
                  <p>Australia</p>
                </div>
              </div>
               <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/india.svg" @click="addToCountry('india')" alt="">
                  <p>India</p>
                </div>
              </div>
               <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/ireland.svg" @click="addToCountry('ireland')" alt="">
                  <p>Ireland</p>
                </div>
              </div>
               <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/kenya.svg" @click="addToCountry('kenya')" alt="">
                  <p>Kenya</p>
                </div>
              </div>
               <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/itali.svg" @click="addToCountry('itali')" alt="">
                  <p>Itali</p>
                </div>
              </div>
               <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/mexico.svg" @click="addToCountry('mexico')" alt="">
                  <p>Mexico</p>
                </div>
              </div>
               <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/SA.svg" @click="addToCountry('SA')" alt="">
                  <p>South Africa</p>
                </div>
              </div>
              <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/nigeria.svg" @click="addToCountry('nigeria')" alt="">
                  <p>Nigeria</p>
                </div>
              </div>
              <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/UAE.svg" @click="addToCountry('UAE')" alt="">
                  <p>UAE</p>
                </div>
              </div>
              <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/US.svg" @click="addToCountry('US')" alt="">
                  <p>US</p>
                </div>
              </div>
              <div class="col-6 col-md-4 col-sm-4 col-lg-4">
                <div class="card">
                  <img src="@/assets/singapore.svg" @click="addToCountry('singapore')" alt="">
                  <p>Singapore</p>

                </div>
              </div>
            </div>
          </div>
          <div class="col-md-2"></div>
        </div>
      </div>
    </div>
    <TheFooter />
  </div>

</template>
<script>
import TheFooter from '@/components/TheFooter';

export default {
  name:'Region',
  data(){
    return{
      countries:[]
    }
  },
  components:{
    TheFooter
  },
  methods:{
    addToCountry(key){
      this.countries.unshift(key)
    }
  }
}
</script>

<<<<<<< HEAD:src/views/general/regionPage/regionPage.vue
<style lang="scss">
  @import './regionPage.scss';
=======
<style lang="scss" scoped="true">
  @import './RegionPage.scss';
>>>>>>> brach out:src/views/consumers/Region/RegionPage.vue
</style>

