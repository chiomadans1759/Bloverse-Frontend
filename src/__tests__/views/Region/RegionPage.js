import { mount, RouterLinkStub } from '@vue/test-utils'
import Region from '@/views/consumers/Region/RegionPage.vue';

describe('Region Page', () => {
  it('should render correctly', () => {

    const wrapper = mount(Region, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.find(RouterLinkStub).props().to).toBe('/')
    expect(wrapper.element).toMatchSnapshot();
  });
});
