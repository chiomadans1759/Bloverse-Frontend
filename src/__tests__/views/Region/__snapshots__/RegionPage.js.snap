// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Region Page should render correctly 1`] = `
<div
  class="region-container"
>
  <div
    class="top"
  >
    <div>
      <span
        class="top-left-container"
      >
        <a
          href="/"
        >
          <i
            class="fal fa-long-arrow-left fa-2x"
          />
           
          <span>
            Back
          </span>
        </a>
         
        <a
          class="router-link"
        >
          <img
            class="logo"
            src="@/assets/Logo.svg"
          />
        </a>
      </span>
    </div>
  </div>
   
  <div
    class="content"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-2"
        />
         
        <div
          class="col-12 col-md-8 col-sm-8 col-lg-8"
        >
          <br />
          <br />
           
          <h3>
            Select Your Favourite Country
          </h3>
           
          <h5>
            Selected 
            <b>
              0
            </b>
             countries
          </h5>
           
          <br />
           
          <div
            class="row"
          >
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/UK.svg"
                />
                 
                <p>
                  UK
                </p>
              </div>
            </div>
             
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/france.svg"
                />
                 
                <p>
                  France
                </p>
              </div>
            </div>
             
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/canada.svg"
                />
                 
                <p>
                  Canada
                </p>
              </div>
            </div>
             
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/germany.svg"
                />
                 
                <p>
                  Germany
                </p>
              </div>
            </div>
             
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/ghana.svg"
                />
                 
                <p>
                  Ghana
                </p>
              </div>
            </div>
             
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/australia.svg"
                />
                 
                <p>
                  Australia
                </p>
              </div>
            </div>
             
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/india.svg"
                />
                 
                <p>
                  India
                </p>
              </div>
            </div>
             
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/ireland.svg"
                />
                 
                <p>
                  Ireland
                </p>
              </div>
            </div>
             
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/kenya.svg"
                />
                 
                <p>
                  Kenya
                </p>
              </div>
            </div>
             
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/itali.svg"
                />
                 
                <p>
                  Itali
                </p>
              </div>
            </div>
             
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/mexico.svg"
                />
                 
                <p>
                  Mexico
                </p>
              </div>
            </div>
             
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/SA.svg"
                />
                 
                <p>
                  South Africa
                </p>
              </div>
            </div>
             
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/nigeria.svg"
                />
                 
                <p>
                  Nigeria
                </p>
              </div>
            </div>
             
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/UAE.svg"
                />
                 
                <p>
                  UAE
                </p>
              </div>
            </div>
             
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/US.svg"
                />
                 
                <p>
                  US
                </p>
              </div>
            </div>
             
            <div
              class="col-6 col-md-4 col-sm-4 col-lg-4"
            >
              <div
                class="card"
              >
                <img
                  alt=""
                  src="@/assets/singapore.svg"
                />
                 
                <p>
                  Singapore
                </p>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="col-md-2"
        />
      </div>
    </div>
  </div>
   
  <footer
    id="the-footer"
  >
    <div
      class="container"
    >
      <div
        class="row"
        id="desktop"
      >
        <div
          class="col"
        >
          <ul
            class="list-inline"
          >
            <li
              class="list-inline-item"
            >
              <a>
                Bloverse
              </a>
            </li>
             
            <li
              class="list-inline-item pl-4"
            >
              <a>
                Terms
              </a>
            </li>
             
            <li
              class="list-inline-item pl-2"
            >
              <a>
                Privacy
              </a>
            </li>
             
            <li
              class="list-inline-item pl-2"
            >
              <a>
                About
              </a>
            </li>
          </ul>
        </div>
         
        <div
          class="col-auto pt-2"
        >
          <p
            id="year-text"
          >
            © 2019 Bloverse. All Right Reserved
          </p>
        </div>
      </div>
       
      <div
        class="row pt-2"
        id="mobile"
      >
        <div
          class="col-8"
        >
          <ul
            class="list-inline"
          >
            <li
              class="list-inline-item pl-4"
              style="font-size: 15px;"
            >
              
            bloverse
          
            </li>
             
            <li
              class="list-inline-item pl-4"
            >
              <a
                id="linkk"
              >
                Terms
              </a>
            </li>
             
            <li
              class="list-inline-item pl-2"
            >
              <a
                id="linkk"
              >
                Privacy
              </a>
            </li>
             
            <li
              class="list-inline-item pl-2"
            >
              <a
                id="linkk"
              >
                About
              </a>
            </li>
          </ul>
        </div>
         
        <div
          class="col-12 pt-1"
        >
          <p
            id="year-text"
          >
            © 2019 Bloverse. All Right Reserved
          </p>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
